### PR TITLE
bpo-41282: Add deprecation warning and docs for distutils (PEP 632)

### PR DESCRIPTION
**Pull request by zooba**
https://bugs.python.org/issue41282

### CODE REVIEW

File: Doc/distutils/index.rst [hunk 1]
```diff
@@ -12,6 +12,13 @@
    :ref:`distributing-index`
       The up to date module distribution documentations
 
+.. note::
+
+   The ``distutils`` namespace has been deprecated and will be removed in
```
Comment (merwok) [inline Doc/distutils/index.rst]: I wonder if «namespace» will be clear to all readers — what do you think of saying `distutils package` or `distutils modules`?

Comment (zooba) [reply Doc/distutils/index.rst]: I went back and forth on all of those, including during the PEP discussions.

"The entire `distutils` package" is about the only other one I'd be happy with, as people really wanted to argue for only deprecating parts of it (even though most of the bits they wanted to "save" already exist elsewhere in the stdlib or supported packages).

Comment (merwok) [reply Doc/distutils/index.rst]: `entire distutils package` sounds good to me.

**Comment by methane** [comment]
I see this warning is emit for `distutils/__init__.py`.

```
/Users/inada-n/work/python/cpython/Lib/distutils/__init__.py:16: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
```

How about adding stacklevel to emit warning for modules importing distutils?

patch:

```
--- a/Lib/distutils/__init__.py
+++ b/Lib/distutils/__init__.py
@@ -16,4 +16,4 @@
 warnings.warn("The distutils package is deprecated and slated for "
               "removal in Python 3.12. Use setuptools or check "
               "PEP 632 for potential alternatives",
-              DeprecationWarning)
+              DeprecationWarning, 2)
```

warning:

```
/Users/inada-n/work/python/cpython/./setup.py:33: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
```

**Pull request merged**